Update Indexer and Dasboard base with Opensearch 2.1.0

c-bordon opened the pull request:
|Related issue|
|---|
|closes https://github.com/wazuh/wazuh-packages/issues/1555|
|closes https://github.com/wazuh/wazuh-packages/issues/1722|
|closes https://github.com/wazuh/wazuh-packages/issues/1723|

<!--
This template reflects sections that must be included in new Pull requests.
Contributions from the community are really appreciated. If this is the case, please add the
"contribution" to properly track the Pull Request.

Please fill the table above. Feel free to extend it at your convenience.
-->

## Description

<!--
Add a clear description of how the problem has been solved.
-->
The adaptation of our packages for the adaptation of Opensearch 2.1.0 is carried out, the changes applied by opensearch in its packages are compared and some of its changes are taken

## Logs example

<!--
Paste here related logs
-->
![Screenshot_20220727_162542](https://user-images.githubusercontent.com/64099752/181355497-a360b529-e4f6-4020-9e63-064d8f963214.png)

## Tests


<!-- Minimum checks required -->
- Build the package in any supported platform
  - [x] Linux
  - [ ] Windows
  - [ ] macOS
  - [ ] Solaris
  - [ ] AIX
  - [ ] HP-UX
- [x] Package installation
- [x] Package upgrade
- [ ] Package downgrade
- [ ] Package remove
- [ ] Package install/remove/install
- [ ] Change added to CHANGELOG.md

<!-- Depending on the affected OS -->
- Tests for Linux RPM
  - [x] Build the package for x86_64
  - [ ] Build the package for i386
  - [ ] Build the package for armhf
  - [ ] Build the package for aarch64
  - [ ] `%files` section is correctly updated if necessary
- Tests for Linux deb
  - [x] Build the package for x86_64
  - [ ] Build the package for i386
  - [ ] Build the package for armhf
  - [ ] Build the package for aarch64
  - [ ] Package install/remove/install
  - [ ] Package install/purge/install
  - [ ] Check file permissions after installing the package
- Tests for macOS
  - [ ] Test the package from macOS Sierra to Mojave
- Tests for Solaris
  - [ ] Test the package on Solaris 10
  - [ ] Test the package on Solaris 11
  - [ ] Check file permissions on Solaris 11 template
- Tests for IBM AIX
  - [ ] `%files` section is correctly updated if necessary
  - [ ] Check the changes from IBM AIX 5 to 7

alberpilot commented on stack/dashboard/base/files/etc/plugins_install_settings.js:
```diff
@@ -0,0 +1,97 @@
+"use strict";
```
Please take into consideration to make `sed` instead of having the entire file because be aware and import automatically changes on this file when building a new base-based future Opensearch versions.

c-bordon replied:
Done: 5edb32d23ad5c015d5d9f257a4e32e4131250714

alberpilot reviewed (changes requested):
Please add upgrade testing

c-bordon commented:
I found a difference in the size of the loading logo, I'm still working on correcting this change since I can't find it:
![Screenshot_20220728_181705](https://user-images.githubusercontent.com/64099752/181639578-84cea2d8-f416-4a06-b637-bc5eaee0827b.png)


- [ ] Upgrade test from 4.3.6 to 4.4.0 package rpm
- [ ] Upgrade test from 4.3.6 to 4.4.0 package deb
- [ ] Upgrade test from 4.4.0 to 4.99.99 package rpm
- [ ] Upgrade test from 4.4.0 to 4.99.99 package deb

c-bordon commented:
Upgrade tests from 4.3.6 to 4.4.0 were successful:
## DEB
### Indexer
```
root@ubuntu20:~# apt install -y ./wazuh-indexer_4.4.0-1_amd64.deb 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Note, selecting 'wazuh-indexer' instead of './wazuh-indexer_4.4.0-1_amd64.deb'
The following packages will be upgraded:
  wazuh-indexer
1 upgraded, 0 newly installed, 0 to remove and 160 not upgraded.
Need to get 0 B/398 MB of archives.
After this operation, 29.1 MB of additional disk space will be used.
Get:1 /root/wazuh-indexer_4.4.0-1_amd64.deb wazuh-indexer amd64 4.4.0-1 [398 MB]
(Reading database ... 143271 files and directories currently installed.)
Preparing to unpack .../wazuh-indexer_4.4.0-1_amd64.deb ...
Unpacking wazuh-indexer (4.4.0-1) over (4.3.6-1) ...
Setting up wazuh-indexer (4.4.0-1) ...
Installing new version of config file /etc/default/wazuh-indexer ...
Installing new version of config file /etc/wazuh-indexer/log4j2.properties ...
Installing new version of config file /etc/wazuh-indexer/opensearch.yml ...
Restarting wazuh-indexer service... OK
Processing triggers for systemd (245.4-4ubuntu3.15) ...
Processing triggers for libc-bin (2.31-0ubuntu9.2) ...
```
### Dashboard
```
root@ubuntu20:~# apt install -y ./wazuh-dashboard_4.4.0-1_amd64.deb 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Note, selecting 'wazuh-dashboard' instead of './wazuh-dashboard_4.4.0-1_amd64.deb'
The following packages will be upgraded:
  wazuh-dashboard
1 upgraded, 0 newly installed, 0 to remove and 160 not upgraded.
Need to get 0 B/142 MB of archives.
After this operation, 71.9 MB of additional disk space will be used.
Get:1 /root/wazuh-dashboard_4.4.0-1_amd64.deb wazuh-dashboard amd64 4.4.0-1 [142 MB]
(Reading database ... 143376 files and directories currently installed.)
Preparing to unpack .../wazuh-dashboard_4.4.0-1_amd64.deb ...
Unpacking wazuh-dashboard (4.4.0-1) over (4.3.6-1) ...
Setting up wazuh-dashboard (4.4.0-1) ...
Installing new version of config file /etc/systemd/system/wazuh-dashboard ...
```

## RPM
### Indexer
```
[root@centos7-1 ~]# yum localinstall -y wazuh-indexer-4.4.0-1.x86_64.rpm 
Failed to set locale, defaulting to C
Loaded plugins: fastestmirror
Examining wazuh-indexer-4.4.0-1.x86_64.rpm: wazuh-indexer-4.4.0-1.x86_64
Marking wazuh-indexer-4.4.0-1.x86_64.rpm as an update to wazuh-indexer-4.3.6-1.x86_64
Resolving Dependencies
--> Running transaction check
---> Package wazuh-indexer.x86_64 0:4.3.6-1 will be updated
---> Package wazuh-indexer.x86_64 0:4.4.0-1 will be an update
--> Finished Dependency Resolution

Dependencies Resolved

=============================================================================================================================================================================================================================================
 Package                                                 Arch                                             Version                                              Repository                                                               Size
=============================================================================================================================================================================================================================================
Updating:
 wazuh-indexer                                           x86_64                                           4.4.0-1                                              /wazuh-indexer-4.4.0-1.x86_64                                           642 M

Transaction Summary
=============================================================================================================================================================================================================================================
Upgrade  1 Package

Total size: 642 M
Downloading packages:
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Updating   : wazuh-indexer-4.4.0-1.x86_64                                                                                                                                                                                              1/2 
  Cleanup    : wazuh-indexer-4.3.6-1.x86_64                                                                                                                                                                                              2/2 
Starting wazuh-indexer service... OK
  Verifying  : wazuh-indexer-4.4.0-1.x86_64                                                                                                                                                                                              1/2 
  Verifying  : wazuh-indexer-4.3.6-1.x86_64                                                                                                                                                                                              2/2 

Updated:
  wazuh-indexer.x86_64 0:4.4.0-1                                                                                                                                                                                                             

Complete!
```

### Dashboard
```
[root@centos7-1 ~]# yum localinstall -y wazuh-dashboard-4.4.0-1.x86_64.rpm 
Failed to set locale, defaulting to C
Loaded plugins: fastestmirror
Examining wazuh-dashboard-4.4.0-1.x86_64.rpm: wazuh-dashboard-4.4.0-1.x86_64
Marking wazuh-dashboard-4.4.0-1.x86_64.rpm as an update to wazuh-dashboard-4.3.6-1.x86_64
Resolving Dependencies
--> Running transaction check
---> Package wazuh-dashboard.x86_64 0:4.3.6-1 will be updated
---> Package wazuh-dashboard.x86_64 0:4.4.0-1 will be an update
--> Finished Dependency Resolution

Dependencies Resolved

=============================================================================================================================================================================================================================================
 Package                                                  Arch                                            Version                                             Repository                                                                Size
=============================================================================================================================================================================================================================================
Updating:
 wazuh-dashboard                                          x86_64                                          4.4.0-1                                             /wazuh-dashboard-4.4.0-1.x86_64                                          645 M

Transaction Summary
=============================================================================================================================================================================================================================================
Upgrade  1 Package

Total size: 645 M
Downloading packages:
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Updating   : wazuh-dashboard-4.4.0-1.x86_64                                                                                                                                                                                            1/2 
  Cleanup    : wazuh-dashboard-4.3.6-1.x86_64                                                                                                                                                                                            2/2 
  Verifying  : wazuh-dashboard-4.4.0-1.x86_64                                                                                                                                                                                            1/2 
  Verifying  : wazuh-dashboard-4.3.6-1.x86_64                                                                                                                                                                                            2/2 

Updated:
  wazuh-dashboard.x86_64 0:4.4.0-1                                                                                                                                                                                                           

Complete!
```

c-bordon commented:
Upgrade tests from 4.4.0 to 99.99.0 were successful:
## DEB
### Indexer
```
root@ubuntu20:~# apt install -y ./wazuh-indexer_99.99.0-1_amd64.deb 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Note, selecting 'wazuh-indexer' instead of './wazuh-indexer_99.99.0-1_amd64.deb'
The following packages will be upgraded:
  wazuh-indexer
1 upgraded, 0 newly installed, 0 to remove and 160 not upgraded.
Need to get 0 B/392 MB of archives.
After this operation, 0 B of additional disk space will be used.
Get:1 /root/wazuh-indexer_99.99.0-1_amd64.deb wazuh-indexer amd64 99.99.0-1 [392 MB]
(Reading database ... 159612 files and directories currently installed.)
Preparing to unpack .../wazuh-indexer_99.99.0-1_amd64.deb ...
Unpacking wazuh-indexer (99.99.0-1) over (4.4.0-1) ...
Setting up wazuh-indexer (99.99.0-1) ...
Restarting wazuh-indexer service... OK
Processing triggers for systemd (245.4-4ubuntu3.15) ...
Processing triggers for libc-bin (2.31-0ubuntu9.2) ...
```
### Dashboard
```
root@ubuntu20:~# apt install -y ./wazuh-dashboard_99.99.0-1_amd64.deb 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Note, selecting 'wazuh-dashboard' instead of './wazuh-dashboard_99.99.0-1_amd64.deb'
The following packages will be upgraded:
  wazuh-dashboard
1 upgraded, 0 newly installed, 0 to remove and 160 not upgraded.
Need to get 0 B/142 MB of archives.
After this operation, 0 B of additional disk space will be used.
Get:1 /root/wazuh-dashboard_99.99.0-1_amd64.deb wazuh-dashboard amd64 99.99.0-1 [142 MB]
(Reading database ... 159612 files and directories currently installed.)
Preparing to unpack .../wazuh-dashboard_99.99.0-1_amd64.deb ...
Unpacking wazuh-dashboard (99.99.0-1) over (4.4.0-1) ...
Setting up wazuh-dashboard (99.99.0-1) ...
```

## RPM
### Indexer
```
[root@centos7-1 ~]# yum localinstall -y wazuh-indexer-99.99.0-1.x86_64.rpm 
Failed to set locale, defaulting to C
Loaded plugins: fastestmirror
Examining wazuh-indexer-99.99.0-1.x86_64.rpm: wazuh-indexer-99.99.0-1.x86_64
Marking wazuh-indexer-99.99.0-1.x86_64.rpm as an update to wazuh-indexer-4.4.0-1.x86_64
Resolving Dependencies
--> Running transaction check
---> Package wazuh-indexer.x86_64 0:4.4.0-1 will be updated
---> Package wazuh-indexer.x86_64 0:99.99.0-1 will be an update
--> Finished Dependency Resolution

Dependencies Resolved

=============================================================================================================================================================================================================================================
 Package                                                Arch                                            Version                                               Repository                                                                Size
=============================================================================================================================================================================================================================================
Updating:
 wazuh-indexer                                          x86_64                                          99.99.0-1                                             /wazuh-indexer-99.99.0-1.x86_64                                          642 M

Transaction Summary
=============================================================================================================================================================================================================================================
Upgrade  1 Package

Total size: 642 M
Downloading packages:
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Updating   : wazuh-indexer-99.99.0-1.x86_64                                                                                                                                                                                            1/2 
  Cleanup    : wazuh-indexer-4.4.0-1.x86_64                                                                                                                                                                                              2/2 
Starting wazuh-indexer service... OK
  Verifying  : wazuh-indexer-99.99.0-1.x86_64                                                                                                                                                                                            1/2 
  Verifying  : wazuh-indexer-4.4.0-1.x86_64                                                                                                                                                                                              2/2 

Updated:
  wazuh-indexer.x86_64 0:99.99.0-1                                                                                                                                                                                                           

Complete!
```

### Dashboard
```
[root@centos7-1 ~]# yum localinstall -y wazuh-dashboard-99.99.0-1.x86_64.rpm 
Failed to set locale, defaulting to C
Loaded plugins: fastestmirror
Examining wazuh-dashboard-99.99.0-1.x86_64.rpm: wazuh-dashboard-99.99.0-1.x86_64
Marking wazuh-dashboard-99.99.0-1.x86_64.rpm as an update to wazuh-dashboard-4.4.0-1.x86_64
Resolving Dependencies
--> Running transaction check
---> Package wazuh-dashboard.x86_64 0:4.4.0-1 will be updated
---> Package wazuh-dashboard.x86_64 0:99.99.0-1 will be an update
--> Finished Dependency Resolution

Dependencies Resolved

=============================================================================================================================================================================================================================================
 Package                                                 Arch                                           Version                                              Repository                                                                 Size
=============================================================================================================================================================================================================================================
Updating:
 wazuh-dashboard                                         x86_64                                         99.99.0-1                                            /wazuh-dashboard-99.99.0-1.x86_64                                         645 M

Transaction Summary
=============================================================================================================================================================================================================================================
Upgrade  1 Package

Total size: 645 M
Downloading packages:
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Updating   : wazuh-dashboard-99.99.0-1.x86_64                                                                                                                                                                                          1/2 
  Cleanup    : wazuh-dashboard-4.4.0-1.x86_64                                                                                                                                                                                            2/2 
  Verifying  : wazuh-dashboard-99.99.0-1.x86_64                                                                                                                                                                                          1/2 
  Verifying  : wazuh-dashboard-4.4.0-1.x86_64                                                                                                                                                                                            2/2 

Updated:
  wazuh-dashboard.x86_64 0:99.99.0-1                                                                                                                                                                                                         

Complete!
```
